golem-worker-service-base/src/service/api_definition.rs

```diff
@@ -141,12 +147,54 @@ impl<Namespace: ApiNamespace, AuthCtx> RegisterApiDefinitionDefault<Namespace, A
             .await?)
     }
 
-    pub async fn register_api(
+    async fn get_all_templates(
         &self,
-        api_definition: &ApiDefinition,
-        key: &ApiDefinitionKey<Namespace>,
-    ) -> Result<(), ApiRegistrationError> {
-        Ok(self.register_repo.register(api_definition, key).await?)
+        definition: &ApiDefinition,
+        auth_ctx: &AuthCtx,
+    ) -> Result<Vec<Template>, ApiRegistrationError> {
+        let get_templates = definition
+            .routes
+            .iter()
+            .cloned()
+            .map(|route| (route.binding.template.clone(), route))
+            .collect::<HashMap<_, _>>()
+            .into_values()
+            .map(|route| async move {
```

Comment:
i would prefer more precise errors:

in this case: if there are 3 routes with same template id
it will report error like Error getting latest template for  route 3.
in my opinion proper error reporting should say:
- Error getting latest template for  route 1.
- Error getting latest template for  route 2.
- Error getting latest template for  route 3.

Reply:
This is how it currently works